app/src/org/commcare/android/database/AndroidDbHelper.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * 
+ */
+package org.commcare.android.database;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.util.Pair;
+
+import net.sqlcipher.database.SQLiteDatabase;
+
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.modern.database.DatabaseHelper;
+import org.commcare.modern.models.EncryptedModel;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.util.externalizable.Externalizable;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
+
+import java.util.HashMap;
+
+/**
+ * Essentially a wrapper around the Java-generic DatabaseHelper
+ * class that allows us to use those function i Android idiomatic classes
+ * (IE ResultSet -> ContentValues, Android Pair instead of generic Pair, etc.
+ *
+ * @author ctsims
+ * @author wspride
+ *
+ */
+public abstract class AndroidDbHelper extends DatabaseHelper {
+    
+    protected Context c;
+    
+    public AndroidDbHelper(Context c) {
+        this.c = c;
+    }
+    
+    public abstract SQLiteDatabase getHandle() throws SessionUnavailableException;
+
+    public ContentValues getContentValues(Externalizable e){
+        ContentValues ret = new ContentValues();
+        HashMap<String, Object> metaFieldsAndValues = DatabaseHelper.getMetaFieldsAndValues(e);
+        for(String key: metaFieldsAndValues.keySet()){
+            Object obj = metaFieldsAndValues.get(key);
```

Comment:
Using `entrySet` in the iterator instead of `keySet` and then a call to `get`is more efficient